ee/bulk-operations-dashboard/api/controllers/software/edit-software.js

```
@@ -0,0 +1,343 @@
+                    receiver__._write = (__newFile, encoding, doneWithThisFile)=>{
+
+                      let FormData = require('form-data');
+                      let form = new FormData();
+                      form.append('team_id', team);
+                      form.append('install_script', installScript);
+                      form.append('post_install_script', postInstallScript);
+                      form.append('pre_install_query', preInstallQuery);
+                      form.append('uninstall_script', uninstallScript);
+                      form.append('software', __newFile, {
+                        filename: software.name,
+                        contentType: 'application/octet-stream'
+                      });
+                      (async ()=>{
+                        try {
+                          await axios.post(`${sails.config.custom.fleetBaseUrl}/api/v1/fleet/software/package`, form, {
+                            headers: {
+                              Authorization: `Bearer ${sails.config.custom.fleetApiToken}`,
+                              ...form.getHeaders()
+                            },
+                          });
+                        } catch(error){
+                          throw new Error('Failed to upload file:'+ require('util').inspect(error, {depth: null}));
+                        }
+                      })()
+                      .then(()=>{
+                        // console.log('ok supposedly a file is finished uploading');
+                        doneWithThisFile();
+                      })
+                      .catch((err)=>{
+                        console.log(err);
+                        doneWithThisFile(err);
+                      });
+                    };//ƒ
+                    return receiver__;
+                  }
+                };
+              },
+            })
+          .intercept((unusedErr)=>{
+            return 'softwareUploadFailed';
```

Rename unusedErr +
```suggestion
            return {'softwareUploadFailed': unusedErr};
```

---

Add a comment about this error catching every error. TODO: add a note about future negotiation of errors and 
1. that I did not test the behavior of uploading a software to a team it was already deployed to to verify that it returns a 409 response.
2. That it currently swallows all errors and returns a softwareUploadFailed response.